### PR TITLE
:sparkles: First stab at RPC server

### PR DESCRIPTION
**Pull request by JonahSussman**
First stab at creating the RPC server

### CODE REVIEW

File: playpen/rpc_server/fake_ide.py [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+from playpen.rpc_server.server import KaiRpcApplication
+
+log = logging.getLogger(__name__)
+
+app = KaiRpcApplication()
+
+
+@app.add_notify(method="logMessage")
+def log_message(app: KaiRpcApplication, message: str, level: str):
+    print(f"{level}: {message}")
```
Comment (djzager) [inline playpen/rpc_server/fake_ide.py:23]: Was going to say that we need this. Nice to see you have it.

File: playpen/rpc_server/main.py [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+
+log = get_logger("jsonrpc")
+
+
+def add_arguments(parser: argparse.ArgumentParser) -> None:
+    parser.description = "Kai RPC Server"
+
+
+def main() -> None:
+    time.sleep(1)
```
Comment (shawn-hurley) [inline playpen/rpc_server/main.py:18]: Why sleep?

File: playpen/rpc_server/rpc.py [hunk 3]
```diff
@@ -0,0 +1,872 @@
...
+        @validate_call(config=ConfigDict(arbitrary_types_allowed=True))
+        @functools.wraps(self.func)
+        def validate_func_args(
+            *args: Any, **kwargs: Any
+        ) -> tuple[tuple[Any, ...], dict[str, Any]]:
+            return args, kwargs
+
+        self.validate_func_args = validate_func_args
+
+        # TODO: Generate docs
```
Comment (shawn-hurley) [inline playpen/rpc_server/rpc.py:354]: We need to remove the commented code or convert to doc string IMO.

**Review by shawn-hurley — APPROVED** [review]
Just comment why the sleep and clean up the code in comments

File: playpen/rpc_server/main.py (hunk 2 above)
Comment (JonahSussman) [reply playpen/rpc_server/main.py:18]: Leftover, removing

**Pull request merged**